Docs: testing and development (#421)
breez/spark-sdk: docs/breez-sdk/src/guide/testing.md

# Testing and Development

## Regtest Network

> For most testing and development, we recommend using the **regtest network** - a deployed test network maintained by Lightspark that provides a safe environment with no real value at risk.

Reviewer: Not 100% sure if network should be capitalized, but I think Regtest should
```suggestion
For most testing and development, we recommend using the **Regtest Network** - a deployed test network maintained by Lightspark that provides a safe environment with no real value at risk.
```

> For most testing and development, we recommend using the **regtest network** - a deployed test network maintained by Lightspark that provides a safe environment with no real value at risk.

Reviewer 2: Suggestion because the SDK is not dangerous to use, even on mainnet.

```suggestion
For most testing and development, we recommend using the **regtest network** - a deployed test network maintained by Lightspark that is free to use and carries no real-world value.
```

Author: Makes total sense, thanks

### What You Can Test on Regtest

- **Spark Payments**: Bitcoin and token payments using the Spark protocol
- **Deposits**: Receiving test Bitcoin from the [Lightspark Regtest Faucet](https://app.lightspark.com/regtest-faucet)
- **Withdrawals**: Sending funds back to on-chain addresses
- **Token Issuance**: Creating and testing tokens using the SDK's [issuing functionality](./issuing_tokens.md)

### Getting Started

1. [Initialize the SDK](./initializing.md) using the default regtest config (no API key required)
2. [Generate a Bitcoin receiving address](./receive_payment.md#bitcoin) 
3. Request funds from the [faucet](https://app.lightspark.com/regtest-faucet) to your generated address
4. Test all Spark-related functionality safely

> ## Lightning Network Testing

Reviewer: ```suggestion
## Lightning Network testing
```

For Lightning payments specifically, we recommend testing on **mainnet with small amounts** since the regtest network doesn't have a developed Lightning Network.

Use real satoshis but keep transaction values very low while verifying payment flows work correctly.

## Development Best Practices

- **Start with regtest** for most development and testing

> - **Use mainnet** for Lightning testing

Reviewer: ```suggestion
- **Use Mainnet** for Lightning testing
```
